R/convertLassaletta2014.R

```diff
@@ -1,50 +1,61 @@
 #' @title convertLassaletta2014
-#' @description converts the dataset of
-#' Lassaletta, L., G. Billen, B. Grizzetti, J. Angalde, and J. Garnier. 2014. 
-#' 50 Year Trends in Nitrogen Use Efficiency of World Cropping Systems: The Relationship between Yield and Nitrogen Input to Cropland.
+#' @description converts the following dataset into a dataset including all countries:
+#' Lassaletta, L., G. Billen, B. Grizzetti, J. Angalde, and J. Garnier. 2014.
+#' 50 Year Trends in Nitrogen Use Efficiency of World Cropping Systems:
+#' The Relationship between Yield and Nitrogen Input to Cropland.
 #' Environmental Research Letters.
-#' into a dataset including all countries. Replacing Soviet Union by Russia and Yugoslavia by Serbia without detailed disaggregation.
 #' @param x data object that gets provided by wrapper function readSource
-#' @param subtype budget provides the nr cropland budgets, fert_to_cropland the sahre of inorganic fertilizers being applied to croplands
+#' @param subtype budget: provides the nr cropland budgets,
+#' fert_to_cropland: the share of inorganic fertilizers being applied to croplands
 #' @return Magpie object with results on country level.
-#' @author Benjamin Leon Bodirsky
+#' @examples
+#' \dontrun{
+#' readSource("Lassaletta2014", convert = TRUE)
+#' }
+#' @author Benjamin Leon Bodirsky, Pascal Führlich
 #' @seealso
 #' [readLassaletta2014()],
 #' [readSource()]
-#' @examples
-#' 
-#' \dontrun{ 
-#' readSource("Lassaletta2014",convert=TRUE)
-#' }
+#' @importFrom madrat toolAggregate toolCountryFill
+#' @importFrom magclass add_columns getItems as.magpie
+convertLassaletta2014 <- function(x, subtype) {
+  if (subtype == "fert_to_cropland") {
```

Comment:
I'm not sure this can be generalized as done here. fert_to_cropland is a share, while budget is an absolute value. this therefore requires different aggregation methods.

Reply:
at least at the point of the toolAggregate method, there should be a differentiation